src/launchConfigManager.ts

```diff
@@ -165,4 +178,41 @@ export class LaunchConfigManager {
     isValidLaunchConfig(): boolean {
         return this.isValidConfig;
     }
+
+    getMissingConfigs(userConfigs: Record<string, unknown>[], extensionConfigs: Record<string, unknown>[]): Record<string, unknown>[] {
+        const missingConfigs: Record<string, unknown>[] = [];
+        for (const extensionConfig of extensionConfigs) {
+            let configExists = false;
+            for (const userConfig of userConfigs) {
+                if (this.compareConfigs(userConfig, extensionConfig)) {
```

Comment:
Because this is effectively a deep comparison, this can end up adding new config entries that mostly duplicate an existing entry if a user has altered a contained value we expect to match (e.g. `--remote-debugging-port`). I see this does allow for additional entries in arrays for things like `runtimeArgs` and allows certain values to be different, though.

The problem occurs when we do hit a non-allowed/incompatible modification, but the name stays the same. In this case end up two entries with the same name after our fixup. Now the compound isn't necessarily going to use the correct one (it appears to pick whichever comes first which is often the incorrect entry). 

I think we want to ensure we also remove/rename the invalid entry in this case.